program_intent_engagement/apps/api/v1/views.py

```diff
@@ -1 +1,61 @@
-# Create your views here.
+"""
+V1 API Views
+"""
+from itertools import groupby
+
+from rest_framework import status
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+
+from program_intent_engagement.apps.core.models import ProgramIntent
+from program_intent_engagement.apps.api.serializers import ProgramIntentSerializer
+
+class MostRecentAndCertainIntentsView(APIView):
+    """
+    Retrieve a list of user intents
+    This endpoint returns a list of intents specified by the user.
+      Grouped by programs and sorted by most recent CERTAIN intent
+    Path: /api/[version]/intents/most-recent-and-certain
+    Returns:
+     * 200: OK
+     * 400: Query returned no results
+    """
+    authentication_classes = (JwtAuthentication,)
+    serializer_class = ProgramIntentSerializer
+
+    def get(self, request):
+        """
+        This view should return the best intents per program
+        for the currently authenticated user.
+
+        Best intent is defined as the most recent CERTAIN intent
+        """
+
+        all_intents = list(ProgramIntent.objects.values().order_by('program_uuid'))
```

Comment:
I believe this will return all program intent objects, regardless of user. A filter clause should be added so this is limited to only the request user